### PR TITLE
[20240701] 맹인호_2주차 LinkedList

### PR DESCRIPTION
**Pull request by MaengSung**

### CODE REVIEW

File: inho/src/main/java/com/dataStructure/LinkedList.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        Node<E> p = head;
+
+        for(int i = 0; i < index; i++){
+            p = p.next;
+        }
+
+        return p;
+    }
+
+    public void addFirst(E e){
```
Comment (MaengSung) [inline inho/src/main/java/com/dataStructure/LinkedList.java:41]: removeFirst 랑 헷갈렸어용

File: inho/src/main/java/com/dataStructure/LinkedList.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    @Override
+    public void add(int index, Object value) {
+        if(index < 0 || index >= size){
+            throw new IndexOutOfBoundsException();
+        }
+
+        if(index == 0){
+            addFirst((E) value);
+        }
+        if(index == size-1){
```
Comment (MaengSung) [inline inho/src/main/java/com/dataStructure/LinkedList.java:84]: size-1 -> size

File: inho/src/main/java/com/dataStructure/LinkedList.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            add(value);
+        }
+        Node<E> p = new Node<>((E) value);
+        Node<E> q = head;
+        for(int i =0; i<index; i++){
```
Comment (MaengSung) [inline inho/src/main/java/com/dataStructure/LinkedList.java:89]: index -> index-1

File: inho/src/main/java/com/dataStructure/LinkedList.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+            q.next = null;
+
+            head = p;
+            size--;
+
+            return element;
+        }else{
+            Node<E> p = head;
+            Node<E> q = head;
+            for(int i =0; i<index-1; i++,q = q.next){
```
Comment (MaengSung) [inline inho/src/main/java/com/dataStructure/LinkedList.java:141]: index 설정

**Comment by koosco** [comment]
고생요

**Pull request merged**